File: src/test/java/com/gridgame/common/ConstantsTest.java
package com.gridgame.common;

import org.junit.Test;
import static org.junit.Assert.*;

public class ConstantsTest {

    @Test
    public void testGridSizeIsPositive() {
        assertTrue(Constants.GRID_SIZE > 0);
    }

    @Test
    public void testCellSizeIsPositive() {
        assertTrue(Constants.CELL_SIZE_PX > 0);
    }

    @Test
    public void testViewportCellsIsPositive() {
        assertTrue(Constants.VIEWPORT_CELLS > 0);
    }

    @Test
    public void testViewportSizeCalculation() {
        assertEquals(Constants.VIEWPORT_CELLS * Constants.CELL_SIZE_PX,
                     Constants.VIEWPORT_SIZE_PX);
    }

    @Test
    public void testPacketSizeIsPositive() {
        assertTrue(Constants.PACKET_SIZE > 0);
    }

    @Test
    public void testServerPortIsValid() {
        assertTrue(Constants.SERVER_PORT > 0);
        assertTrue(Constants.SERVER_PORT < 65536);
    }

    @Test
    public void testHeartbeatIntervalIsPositive() {
        assertTrue(Constants.HEARTBEAT_INTERVAL_MS > 0);
    }

    @Test
    public void testClientTimeoutIsPositive() {
        assertTrue(Constants.CLIENT_TIMEOUT_MS > 0);
    }

    @Test
    public void testTimeoutGreaterThanHeartbeat() {
        assertTrue(Constants.CLIENT_TIMEOUT_MS > Constants.HEARTBEAT_INTERVAL_MS);
    }

    @Test
    public void testMoveRateLimitIsPositive() {
        assertTrue(Constants.MOVE_RATE_LIMIT_MS > 0);
    }
}

File: src/test/scala/com/gridgame/common/ConstantsTest.scala
package com.gridgame.common

import org.junit.Test
import org.junit.Assert._

class ConstantsTest {

  @Test
  def testGridSizeIsPositive(): Unit = {
    assertTrue(Constants.GRID_SIZE > 0)
  }

  @Test
  def testCellSizeIsPositive(): Unit = {
    assertTrue(Constants.CELL_SIZE_PX > 0)
  }

  @Test
  def testViewportCellsIsPositive(): Unit = {
    assertTrue(Constants.VIEWPORT_CELLS > 0)
  }

  @Test
  def testViewportSizeCalculation(): Unit = {
    assertEquals(Constants.VIEWPORT_CELLS * Constants.CELL_SIZE_PX, Constants.VIEWPORT_SIZE_PX)
  }

  @Test
  def testPacketSizeIsPositive(): Unit = {
    assertTrue(Constants.PACKET_SIZE > 0)
  }

  @Test
  def testServerPortIsValid(): Unit = {
    assertTrue(Constants.SERVER_PORT > 0)
    assertTrue(Constants.SERVER_PORT < 65536)
  }

  @Test
  def testHeartbeatIntervalIsPositive(): Unit = {
    assertTrue(Constants.HEARTBEAT_INTERVAL_MS > 0)
  }

  @Test
  def testClientTimeoutIsPositive(): Unit = {
    assertTrue(Constants.CLIENT_TIMEOUT_MS > 0)
  }

  @Test
  def testTimeoutGreaterThanHeartbeat(): Unit = {
    assertTrue(Constants.CLIENT_TIMEOUT_MS > Constants.HEARTBEAT_INTERVAL_MS)
  }

  @Test
  def testMoveRateLimitIsPositive(): Unit = {
    assertTrue(Constants.MOVE_RATE_LIMIT_MS > 0)
  }
}
